pkg/apis/nmstate/v1alpha1/nodenetworkconfigurationpolicy_types.go

```diff
@@ -0,0 +1,74 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// NodeNetworkConfigurationPolicyMatch define the matching criteria to apply
+// the policy
+// +k8s:openapi-gen=true
+type NodeNetworkConfigurationPolicyMatch struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+
+	// Node names to apply the policy
+	Nodes []string `json:"nodes,omitempty"`
```

Comment:
Let's stick to existing kubernetes API and use nodeSelector: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector

It does matching based on labels, so it is pretty flexible.

Reply:
ditto